osu.Game/OsuGame.cs

```diff
@@ -426,6 +427,12 @@ public void PresentScore(ScoreInfo score, ScorePresentType presentType = ScorePr
             }, validScreens: new[] { typeof(PlaySongSelect) });
         }
 
+        public override Task Import(Stream stream, string filename)
+        {
+            WaitForReady(() => this, _ => Task.Run(() => base.Import(stream, filename)));
+            return Task.CompletedTask;
+        }
```

Comment:
This override looks unsafe to await on. If I'm reading this correct, the base `Import(Stream, string)` method, if awaited on, won't return control to the caller until the import sequence has actually completed, while this variant is essentially a fire-and-forget: `WaitForReady` may execute the action immediately, but it might also schedule it; worse yet, the action is wrapped in a `Task.Run` anyway. Therefore I imagine this will always return to the caller immediately after the task is started on/assigned to a pool thread.

I think what you might actually want here is to assign the inner task to a variable, pass it to `WaitForReady` so that it can be executed on another thread by `Task.Run`, and then return that stored-but-unstarted task so that its state changes can be observed by callers.

Reply:
Are we ever gonna await on this though? Looking at the other place we are are calling the `Import(string[])`  overload,  we are just firing the import and forgetting about it since the UI logic / error logic is contained further down in the call hierarchy.

Reply:
Actuall, nevermind , the approach given above sounds more logical given we're scheduling the task to be long running.